glean-core/csharp/Glean/Glean.cs

```diff
@@ -104,6 +105,9 @@ string dataDir
                     return;
                 }
 
+                // Registering pings needs to be after Glean initialization.
+                RegisterPings();
```

Comment:
It shouldn't: the way we use the dispatchers should guarantee the proper execution model. Can you kindly try to move it back where it was? :) Maybe this is not even needed in this language.

Reply:
I think I get the sense how to make it work. It looks like I can't just make `GleanInternalPings` to be a static class and its pings are static inline.

Because I need a place for `GleanInternalPings` to instance its attributes. Therefore, I still choose to make 'GleanInternalPings' as a lazy instantiation singleton. Currently, I only make its pings are inline. Let me know if you feel lazy instance them is better.